// @wagmi/vue@0.0.26/dist/esm/exports/index.js
////////////////////////////////////////////////////////////////////////////////
// Plugin
////////////////////////////////////////////////////////////////////////////////
// biome-ignore lint/performance/noBarrelFile: entrypoint module
export { configKey, WagmiPlugin } from '../plugin.js';
////////////////////////////////////////////////////////////////////////////////
// Errors
////////////////////////////////////////////////////////////////////////////////
export { BaseError } from '../errors/base.js';
export { WagmiPluginNotFoundError, WagmiInjectionContextError, } from '../errors/plugin.js';
////////////////////////////////////////////////////////////////////////////////
// Composables
////////////////////////////////////////////////////////////////////////////////
export { useAccount, } from '../composables/useAccount.js';
export { useAccountEffect, } from '../composables/useAccountEffect.js';
export { useBalance, } from '../composables/useBalance.js';
export { useBlockNumber, } from '../composables/useBlockNumber.js';
export { useChainId, } from '../composables/useChainId.js';
export { useClient, } from '../composables/useClient.js';
export { useConnectorClient, } from '../composables/useConnectorClient.js';
export { useChains, } from '../composables/useChains.js';
export { useConfig, } from '../composables/useConfig.js';
export { useConnect, } from '../composables/useConnect.js';
export { useConnections, } from '../composables/useConnections.js';
export { useConnectors, } from '../composables/useConnectors.js';
export { useDisconnect, } from '../composables/useDisconnect.js';
export { useEnsAddress, } from '../composables/useEnsAddress.js';
export { useEnsAvatar, } from '../composables/useEnsAvatar.js';
export { useEnsName, } from '../composables/useEnsName.js';
export { useEstimateGas, } from '../composables/useEstimateGas.js';
export { useReadContract, } from '../composables/useReadContract.js';
export { useReconnect, } from '../composables/useReconnect.js';
export { useSendTransaction, } from '../composables/useSendTransaction.js';
export { useSignMessage, } from '../composables/useSignMessage.js';
export { useSignTypedData, } from '../composables/useSignTypedData.js';
export { useSimulateContract, } from '../composables/useSimulateContract.js';
export { useSwitchAccount, } from '../composables/useSwitchAccount.js';
export { useSwitchChain, } from '../composables/useSwitchChain.js';
export { useTransaction, } from '../composables/useTransaction.js';
export { useTransactionReceipt, } from '../composables/useTransactionReceipt.js';
export { useWatchBlockNumber, } from '../composables/useWatchBlockNumber.js';
export { useWaitForTransactionReceipt, } from '../composables/useWaitForTransactionReceipt.js';
export { useWriteContract, } from '../composables/useWriteContract.js';
////////////////////////////////////////////////////////////////////////////////
// @wagmi/core
////////////////////////////////////////////////////////////////////////////////
export { createConfig, createConnector, ChainNotConfiguredError, ConnectorAlreadyConnectedError, ConnectorNotFoundError, ConnectorAccountNotFoundError, ConnectorChainMismatchError, ProviderNotFoundError, SwitchChainNotSupportedError, createStorage, noopStorage, 
// Transports
custom, fallback, http, webSocket, unstable_connector, 
// Utilities
cookieStorage, cookieToInitialState, deepEqual, deserialize, normalizeChainId, parseCookie, serialize, } from '@wagmi/core';
////////////////////////////////////////////////////////////////////////////////
// Version
////////////////////////////////////////////////////////////////////////////////
export { version } from '../version.js';
